File: app/src/main/java/uk/co/marksheehan/doomtowndeckbuilder/CardAdapter.java
package uk.co.marksheehan.doomtowndeckbuilder;


import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.example.mark.doomtowndeckbuilder.R;

import java.util.List;

public class CardAdapter extends RecyclerView.Adapter<CardAdapter.CardViewHolder>
{
    public CardAdapter(List<CardModel> list)
    {
        mItemList = list;
    }

    private List<CardModel> mItemList;

    public class CardViewHolder extends RecyclerView.ViewHolder
    {
        public TextView cost;
        public TextView cardName;

        public CardViewHolder(View itemView)
        {
            super(itemView);
            cardName = itemView.findViewById(R.id.cardName);
            cost = itemView.findViewById(R.id.cost);
        }
    }

    @Override
    public CardViewHolder onCreateViewHolder(ViewGroup parent, int viewType)
    {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.card, parent, false);
        CardViewHolder viewHolder = new CardViewHolder(itemView);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(CardViewHolder holder, int position)
    {
        final CardModel currentCard = mItemList.get(position);

        String cost = "" + currentCard.cost;

        holder.cardName.setText(currentCard.title);
        holder.cost.setText(cost);
    }

    @Override
    public int getItemCount() {
        return mItemList.size();
    }
}

File: app/src/main/java/net/marksheehan/doomtowndeckbuilder/adapters/CardAdapter.kt
package net.marksheehan.doomtowndeckbuilder.adapters

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.recyclerview.widget.RecyclerView
import com.squareup.picasso.Picasso
import net.marksheehan.doomtowndeckbuilder.R
import net.marksheehan.doomtowndeckbuilder.datamodel.CardModel

class CardAdapter(private val itemList: List<CardModel>, val onCardModelClicked : (CardModel)-> Unit) : RecyclerView.Adapter<CardAdapter.CardViewHolder>() {

    inner class CardViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var cardImage: ImageView = itemView.findViewById(R.id.cardImage)

        fun bind(cardModel: CardModel){
            itemView.setOnClickListener{onCardModelClicked(cardModel)}
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CardViewHolder {
        val itemView = LayoutInflater.from(parent.context).inflate(R.layout.card_small_view, parent, false)
        return CardViewHolder(itemView)
    }

    override fun onBindViewHolder(holder: CardViewHolder, position: Int) {
        val currentCard = itemList[position]
        holder.bind(currentCard)
        Picasso.get().load(currentCard.getImagePath()).placeholder(R.drawable.card_back).error(R.drawable.card_back).into(holder.cardImage)
    }

    override fun getItemCount(): Int {
        return itemList.size
    }
}
